src/Model/FsTreeNode.re

```diff
@@ -26,49 +34,48 @@ let rec countExpandedSubtree =
 
   | _ => 1;
 
-let file = (path, ~id, ~icon) => {
+let file = (path, ~icon) => {
   let basename = Filename.basename(path);
 
   {
-    id,
     path,
+    hash: _hash(basename),
     displayName: basename,
-    hash: Hashtbl.hash(basename),
     icon,
     kind: File,
     expandedSubtreeSize: 1,
   };
 };
 
-let directory = (~isOpen=false, path, ~id, ~icon, ~children) => {
+let directory = (~isOpen=false, path, ~icon, ~children) => {
   let kind = Directory({isOpen, children});
   let basename = Filename.basename(path);
 
   {
-    id,
     path,
+    hash: _hash(basename),
     displayName: basename,
-    hash: Hashtbl.hash(basename),
     icon,
     kind,
     expandedSubtreeSize: countExpandedSubtree(kind),
   };
 };
 
-let findNodesByLocalPath = (path, tree) => {
-  let pathHashes =
-    path
-    |> String.split_on_char(Filename.dir_sep.[0])
-    |> List.map(Hashtbl.hash);
+let equals = (a, b) => a.hash == b.hash && a.path == b.path;
```

Comment:
Hashes aren't guaranteed to be unique, so use this instead of comparing hashes directly, and especially instead of comparing nodes structurally.